### PR TITLE
feat: update shipment api

### PR DESCRIPTION
**Pull request by rikardq**
This enables a shipment to be updated with external reference id and units.
Further iterations will come.

### CODE REVIEW

File: proto/einride/saga/extend/book/v1beta1/shipment_service.proto [hunk 1]
```diff
@@ -170,6 +171,26 @@ service ShipmentService {
       }
     };
   }
+
+  // Update a shipment in a space.
+  //
+  // Currently only allows updating shipment-reference-id and units
```
Comment (DavyJ0nes) [inline proto/einride/saga/extend/book/v1beta1/shipment_service.proto:177]: should this read;
```suggestion
  // Currently only allows updating external_reference_id and units
```

File: proto/einride/saga/extend/book/v1beta1/shipment_service.proto [hunk 2]
```diff
@@ -170,6 +171,26 @@ service ShipmentService {
       }
     };
   }
+
+  // Update a shipment in a space.
+  //
+  // Currently only allows updating external_reference_id and units
+  // For units you cannot update individual units, so you MUST provide all the other units
```
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/shipment_service.proto:178]:
```suggestion
  // For units you cannot update individual units, so you MUST provide all the units
```
or
```suggestion
  // For units you cannot update individual units, so you MUST provide all the units as well.
```

File: proto/einride/saga/extend/book/v1beta1/shipment_service.proto [hunk 3]
```diff
@@ -170,6 +171,26 @@ service ShipmentService {
...
+  // This is an AIP standard [Create](https://google.aip.dev/134) method.
```
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/shipment_service.proto:179]:
```suggestion
  // This is an AIP standard [Update](https://google.aip.dev/134) method.
```

File: proto/einride/saga/extend/book/v1beta1/shipment_service.proto (hunk 2 above)
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/shipment_service.proto:178]: I suggest removing this and just having this information down on the update mask.

File: proto/einride/saga/extend/book/v1beta1/shipment_service.proto [hunk 4]
```diff
@@ -210,3 +231,16 @@ message ReleaseShipmentRequest {
...
   ];
 }
+
+// Request to UpdateShipment method.
+message UpdateShipmentRequest {
+  // The resource which replaces the current resource.
+  Shipment shipment = 1 [(google.api.field_behavior) = REQUIRED];
+  // The update mask applies to the shipment.
+  // This must be set. Only external_reference_id, units and sender are currently supported.
+  // as well as the updated one.
```
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/shipment_service.proto:241]:
```suggestion
 
```
This row can be removed.

**Review by gostajonasson — APPROVED** [review]
:star2:

**Pull request merged**